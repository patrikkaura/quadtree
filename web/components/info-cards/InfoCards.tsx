import { QuadtreeData } from "@components/types";
import Stack from "@mui/material/Stack";
import { useMemo } from "react";

import { Card } from "./Card";

export type Props = {
  threshold: number;
  stats: QuadtreeData["stats"];
};

export function InfoCards({ threshold, stats }: Props) {
  const {
    sequenceLength,
    frequencyPerThousand,
    gcCount,
    numberOfQuadruplexes,
  } = stats;

  const cardData = useMemo(
    () => [
      {
        header: "Sequence length",
        content: sequenceLength,
        width: 200,
        color: "#98DDCA",
      },
      {
        header: "Threshold",
        content: threshold,
        width: 200,
        color: "#98DD99",
      },
      {
        header: "Frequency per thousand",
        content: frequencyPerThousand,
        width: 200,
        color: "#D5ECC2",
      },
      { header: "GC count", content: gcCount, width: 200, color: "#FFD3B4" },
      {
        header: "Number of quadruplexes",
        content: numberOfQuadruplexes,
        width: 200,
        color: "#FFAAA7",
      },
    ],
    [
      sequenceLength,
      threshold,
      frequencyPerThousand,
      gcCount,
      numberOfQuadruplexes,
    ]
  );

  return (
    <Stack direction="row" spacing={2}>
      {cardData.map((data, index) => (
        <Card key={index} {...data} />
      ))}
    </Stack>
  );
}
