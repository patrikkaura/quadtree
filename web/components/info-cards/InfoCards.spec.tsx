import { render } from "@testing-library/react";

import { InfoCards, Props } from "./InfoCards";

describe("InfoCards", () => {
  let props: Props = {
    threshold: 0.5,
    stats: {
      sequenceLength: 100,
      frequencyPerThousand: "10",
      gcCount: "10",
      numberOfQuadruplexes: 5,
    },
  };

  it("should render", () => {
    const { container } = render(<InfoCards {...props} />);

    expect(container).toMatchSnapshot();
  });
});
