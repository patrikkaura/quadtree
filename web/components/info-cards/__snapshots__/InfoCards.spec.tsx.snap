// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoCards should render 1`] = `
<div>
  <div
    class="css-e53awj-MuiStack-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      style="width: 200px; border-top: 5px solid; border-color: #98ddca;"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-16m3jxf-MuiTypography-root"
        >
          Sequence length
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ll0ml8-MuiTypography-root"
        >
          100
        </p>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      style="width: 200px; border-top: 5px solid; border-color: #98dd99;"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-16m3jxf-MuiTypography-root"
        >
          Threshold
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ll0ml8-MuiTypography-root"
        >
          0.5
        </p>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      style="width: 200px; border-top: 5px solid; border-color: #d5ecc2;"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-16m3jxf-MuiTypography-root"
        >
          Frequency per thousand
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ll0ml8-MuiTypography-root"
        >
          10
        </p>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      style="width: 200px; border-top: 5px solid; border-color: #ffd3b4;"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-16m3jxf-MuiTypography-root"
        >
          GC count
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ll0ml8-MuiTypography-root"
        >
          10
        </p>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      style="width: 200px; border-top: 5px solid; border-color: #ffaaa7;"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-16m3jxf-MuiTypography-root"
        >
          Number of quadruplexes
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ll0ml8-MuiTypography-root"
        >
          5
        </p>
      </div>
    </div>
  </div>
</div>
`;
